docs/interactions/Slash_Commands.md

```diff
@@ -0,0 +1,841 @@
+                    "description": "Get permissions for a user",
+                    "type": 1 // 1 is type SUB_COMMAND
+                },
+                {
+                    "name": "edit",
+                    "description": "Edit permissions for a user",
+                    "type": 1
+                }
+            ]
+        },
+        {
+            "name": "role",
+            "description": "Get or edit permissions for a role",
+            "type": 2,
+            "options": [
+                {
+                    "name": "get",
+                    "description": "Get permissions for a role",
+                    "type": 1
+                },
+                {
+                    "name": "edit",
+                    "description": "Edit permissions for a role",
+                    "type": 1
+                }
+            ]
+        }
+    ]
+}
+```
+
+![A command with grouped subcommands. It says /permissions user get](command-with-groups-subcommands.png)
+
+Now, we need some arguments! If we chose `user`, we need to be able to pick a user; if we chose `role`, we need to be able to pick a role. We also want to be able to pick between guild-level permissions and channel-specific permissions. For that, we can use optional arguments:
+
+```js
+{
+    "name": "permissions",
+    "description": "Get or edit permissions for a user or a role",
+    "options": [
+        {
+            "name": "user",
+            "description": "Get or edit permissions for a user",
+            "type": 2, // 2 is type SUB_COMMAND_GROUP
+            "options": [
+                {
+                    "name": "get",
+                    "description": "Get permissions for a user",
+                    "type": 1, // 1 is type SUB_COMMAND
+                    "options": [
+                        {
+                            "name": "user",
+                            "description": "The user to get",
+                            "type": 6, // 6 is type USER
+                            "required": true
+                        },
+                        {
+                            "name": "channel",
+                            "description": "The channel permissions to get",
+                            "type": 7, // 7 is type CHANNEL
+                            "required": false
+                        }
+                    ]
+                },
+                {
+                    "name": "edit",
+                    "description": "Edit permissions for a user",
+                    "type": 1,
+                    "options": [
+                        {
+                            "name": "user",
+                            "description": "The user to edit",
+                            "type": 6,
+                            "required": true
+                        },
+                        {
+                            "name": "channel",
+                            "description": "The channel permissions to edit",
```

Comment:
Maybe the `description` should mention that if it is left out, the guild permissions will be displayed?